go/extractor/cli/go-autobuilder/go-autobuilder.go

```diff
@@ -719,82 +719,150 @@ func installDependenciesAndBuild() {
+			v.goEnvVersion + ") is outside of the supported range (" + minGoVersion + "-" +
+			maxGoVersion + "). Writing an environment file specifying the maximum supported " +
+			"version of Go (" + maxGoVersion + ")."
+		version = maxGoVersion
+		diagnostics.EmitNoGoModAndGoEnvUnsupported(msg)
+	} else {
+		// The version of Go that is installed is supported. We have no indication which version
+		// was intended to be used to build this project. We assume that the installed version is
+		// suitable and do not install a version of Go.
+		msg = "No `go.mod` file found. Version " + v.goEnvVersion + " installed in the " +
+			"environment is supported. Writing an environment file not specifying any " +
+			"version of Go."
 		version = ""
-		diagnostics.EmitUnsupportedVersionEnvironment(msg)
+		diagnostics.EmitNoGoModAndGoEnvSupported(msg)
 	}
 
 	return msg, version
 }
 
-// Check if either `v.goEnvVersionFound` or `v.goModVersionFound` are false. If so, emit
-// a diagnostic and return the version to install, or the empty string if we should not attempt to
-// install a version of Go. We assume that `checkForUnsupportedVersions` has already been
-// called, so any versions that are found are within the supported range.
-func checkForVersionsNotFound(v versionInfo) (msg, version string) {
-	if !v.goEnvVersionFound && !v.goModVersionFound {
-		msg = "No version of Go installed and no `go.mod` file found. Writing an environment " +
-			"file specifying the maximum supported version of Go (" + maxGoVersion + ")."
-		version = maxGoVersion
-		diagnostics.EmitNoGoModAndNoGoEnv(msg)
-	}
+// Assuming `v.goModVersion` is above the supported range, emit a diagnostic and return the
+// version to install, or the empty string if we should not attempt to install a version of Go.
+func getVersionWhenGoModVersionTooHigh(v versionInfo) (msg, version string) {
+	// The project is intended to be built with a version of Go that is above the supported
+	// range. We do not install a version of Go.
+	msg = "The version of Go found in the `go.mod` file (" + v.goModVersion +
+		") is above the supported range (" + minGoVersion + "-" + maxGoVersion +
+		"). Writing an environment file not specifying any version of Go."
+	version = ""
+	diagnostics.EmitGoModVersionTooHigh(msg)
 
-	if !v.goEnvVersionFound && v.goModVersionFound {
-		msg = "No version of Go installed. Writing an environment file specifying the version " +
-			"of Go found in the `go.mod` file (" + v.goModVersion + ")."
-		version = v.goModVersion
-		diagnostics.EmitNoGoEnv(msg)
-	}
+	return msg, version
+}
 
-	if v.goEnvVersionFound && !v.goModVersionFound {
-		msg = "No `go.mod` file found. Version " + v.goEnvVersion + " installed in the " +
-			"environment. Writing an environment file not specifying any version of Go."
+// Assuming `v.goModVersion` is above the supported range, emit a diagnostic and return the
+// version to install, or the empty string if we should not attempt to install a version of Go.
```

Comment:
The comment here is the same as for `getVersionWhenGoModVersionTooHigh` but should really read:
```suggestion
// Assuming `v.goModVersion` is below the supported range, emit a diagnostic and return the
// version to install, or the empty string if we should not attempt to install a version of Go.
```

Reply:
This seems to be the same as the comment above. I'm not sure how that's possible.